pkg/build/controller/factory/factory.go

```diff
@@ -14,16 +14,17 @@ import (
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/watch"
 
 	buildapi "github.com/openshift/origin/pkg/build/api"
+	buildclient "github.com/openshift/origin/pkg/build/client"
 	controller "github.com/openshift/origin/pkg/build/controller"
 	strategy "github.com/openshift/origin/pkg/build/controller/strategy"
-	buildutil "github.com/openshift/origin/pkg/build/util"
 	osclient "github.com/openshift/origin/pkg/client"
 	imageapi "github.com/openshift/origin/pkg/image/api"
 )
 
 type BuildControllerFactory struct {
-	Client              *osclient.Client
-	KubeClient          *kclient.Client
+	OSClient            osclient.Interface
+	KubeClient          kclient.Interface
+	BuildUpdater        buildclient.BuildUpdateInterface
```

Comment:
Again, `buildclient.BuilderUpdater` would make me feel like you want me to be happy.